### PR TITLE
Authorization checker component

### LINKED ISSUE #77: Authorization Checker Component
Page Url: any
Module: Authorization
Authority:
- any
Components: 
- [ ] Authorization Component (onlyCCHead, onlyAdmin, AdminAndEventHead, AdminAndEventCrew)

Usage: conditional rendering based on authorization 

- [ ] Development
- [ ] Unit Test

### PR DESCRIPTION
**Pull request by kachun333**
resolve #77

### CODE REVIEW

**Review by HowardYaw — DISMISSED** [review]

File: src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      eventRole,
+      fallbackEl,
+      eventId,
+      isAuthenticated,
+      isCurrentCCHead,
+      isCurrentAdministrator,
+      isEventHead,
+      isEventCrew
+    } = this.props;
+    const fallbackElement: React.ReactElement | null = fallbackEl ? fallbackEl : null;
```
Comment (junzai97) [inline src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx]: Can explain this line of code?

File: src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if (hasEventRole && hasEventId) {
+      const isOnlyEventHeadCanView = eventRole === EventRole.HEAD;
+      const isOnlyEventCrewCanView = eventRole === EventRole.CREW;
+      const hasPassEventHeadCheck = isOnlyEventHeadCanView && isEventHead;
+      const hasPassEventCrewCheck = isOnlyEventCrewCanView && isEventCrew;
+      hasPassEventRoleChecking = hasPassEventHeadCheck || hasPassEventCrewCheck;
+    }
+    const canRender = hasPassCCRoleChecking || hasPassEventRoleChecking;
+    if (!canRender) return fallbackElement;
+    return children;
```
Comment (junzai97) [inline src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx]: Maybe can simplified into return canRender? fallbackElement: children?

Comment (kachun333) [reply src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx]: fixed in c46b7ec97d7da4f9b39f77bc0fa237fe73f54a0e

File: src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx (hunk 1 above)
Comment (kachun333) [reply src/main/webapp/app/shared/components/authorization-checker/authorization-checker.tsx]: fixed in c46b7ec97d7da4f9b39f77bc0fa237fe73f54a0e

**Review by HowardYaw — APPROVED** [review]

**Review by junzai97 — APPROVED** [review]

**Pull request merged**